test/performance/playwright.config.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * External dependencies
+ */
+import os from 'os';
+import path from 'path';
+import { fileURLToPath } from 'url';
+import { defineConfig, devices } from '@playwright/test';
+
+process.env.WP_ARTIFACTS_PATH ??= path.join( process.cwd(), 'artifacts' );
+process.env.STORAGE_STATE_PATH ??= path.join(
+	process.env.WP_ARTIFACTS_PATH,
+	'storage-states/admin.json'
+);
+process.env.ASSETS_PATH = path.join( __dirname, 'assets' );
+
+const config = defineConfig( {
+	reporter: process.env.CI
+		? undefined // We're using another reporter in CI.
+		: [ [ 'list' ], [ './config/performance-reporter.ts' ] ],
+	forbidOnly: !! process.env.CI,
+	fullyParallel: false,
+	workers: 1,
+	retries: 0,
+	timeout: parseInt( process.env.TIMEOUT || '', 10 ) || 100_000, // Defaults to 100 seconds.
+	testDir: fileURLToPath( new URL( './specs', 'file:' + __filename ).href ),
+	outputDir: path.join( process.env.WP_ARTIFACTS_PATH, 'test-results' ),
+	snapshotPathTemplate:
+		'{testDir}/{testFileDir}/__snapshots__/{arg}-{projectName}{ext}',
+	globalSetup: fileURLToPath(
+		new URL( './config/global-setup.ts', 'file:' + __filename ).href
+	),
+	use: {
+		baseURL: process.env.WP_BASE_URL || 'http://localhost:8889',
+		headless: true,
+		viewport: {
+			width: 960,
+			height: 700,
+		},
+		ignoreHTTPSErrors: true,
+		locale: 'en-US',
+		contextOptions: {
+			reducedMotion: 'reduce',
+			strictSelectors: true,
+		},
+		storageState: process.env.STORAGE_STATE_PATH,
+		actionTimeout: 10_000, // 10 seconds.
+		trace: 'retain-on-failure',
+		screenshot: 'only-on-failure',
+		video: 'on-first-retry',
```

Comment:
Since we don't retry this has no effects. Probably better to just set it to `off`.

Reply:
Now that I think about it, videos can be really helpful for debugging. For example, for a long time, CI was executing the Site Editor Typing test differently than the local env, which was really hard to debug without the visuals. Would it make sense to record failures, or is the trace.zip just enough?

Reply:
I agree. The videos aren't beneficial when debugging the Playwright tests; most of the time, recordings show how tests pass. It's much easier to debug with traces and trace viewer.